roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,113 @@
+# pylint: disable=missing-docstring
+import json
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DockerStorage(OpenShiftCheck):
+    """Check Docker storage sanity.
+
+    This check ensures that thinpool usage is not
+    close to 100% for a containerized installation.
+    """
+
+    name = "docker_storage"
+    tags = ["preflight"]
+
+    max_thinpool_data_usage_percent = 90.0
+    max_thinpool_metadata_usage_percent = 90.0
+
+    @classmethod
+    def is_active(cls, task_vars):
+        """Skip non-containerized installations."""
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        return super(DockerStorage, cls).is_active(task_vars) and is_containerized
+
+    def run(self, tmp, task_vars):
+        self.max_thinpool_data_usage_percent = get_var(task_vars,
+                                                       "max_thinpool_data_usage_percent",
+                                                       default=self.max_thinpool_data_usage_percent)
+        self.max_thinpool_metadata_usage_percent = get_var(task_vars, "max_thinpool_metadata_usage_percent",
+                                                           default=self.max_thinpool_metadata_usage_percent)
+
+        failed, msg = self.check_thinpool_usage(task_vars)
+        if failed:
+            return {"failed": True, "msg": msg}
+
+        return {"changed": False}
+
+    def check_thinpool_usage(self, task_vars):
+        no_data_msg = "no thinpool usage data returned by the host."
+
+        failed, data, msg = self.get_lvs_data(task_vars)
+        if failed:
+            if not msg:
+                msg = no_data_msg
+            return failed, msg
+
+        lv = self.extract_thinpool_obj(data)
+        if not lv:
+            return True, no_data_msg
+
+        failed, data_percent = self.get_thinpool_data_usage(lv)
+        if failed:
+            return failed, no_data_msg
+
+        failed, metadata_percent = self.get_thinpool_metadata_usage(lv)
+        if failed:
+            return failed, no_data_msg
+
+        if data_percent > float(self.max_thinpool_data_usage_percent):
+            msg = "thinpool data usage above maximum threshold of {threshold}%"
+            return True, msg.format(threshold=self.max_thinpool_data_usage_percent)
+
+        if metadata_percent > float(self.max_thinpool_metadata_usage_percent):
+            msg = "thinpool metadata usage above maximum threshold of {threshold}%"
+            return True, msg.format(threshold=self.max_thinpool_metadata_usage_percent)
+
+        return False, ""
+
+    def get_lvs_data(self, task_vars):
+        lvs_cmd = "/sbin/lvs --select vg_name=docker --select lv_name=docker-pool --report-format json"
+        result = self.exec_cmd(lvs_cmd, task_vars)
+        failed = False
+
+        if result.get("failed", False):
+            return True, {}, result.get("msg", "")
```

Comment:
This method returns 3 non-obvious values and they do undocumented...
I'm guessing "did it fail?", "the lvs data I asked for" and "some error message if any". It's a weird API to have failed as the first argument, and might even be unnecessary to have it.

Alternatively, these methods could return what they are meant to return, and if not raise an exception, that would remove a lot of the conditionals.